### PR TITLE
Add doc for TS_VERIFY_CTX_set_certs()

### PR DESCRIPTION
**Pull request by InfoHunter**
This addition is based on PR #9472 

This will definitely fail the test (doc-nits check) since #9472 has not been merged yet.

##### Checklist

- [x] documentation is added or updated

### CODE REVIEW

**Comment by InfoHunter** [comment]
Not sure if the document of this function (or other TS related ones) should be back ported to 1.1.1. It might be worth doing so from my perspective, since 1.1.1 is a LTS release.

**Comment by mspncp** [comment]
Adding missing documentation for existing functions qualifies for backporting just like fixing a bug.

File: doc/man3/TS_VERIFY_CTX_set_certs.pod [hunk 1]
```diff
@@ -0,0 +1,44 @@
+=pod
+
+=head1 NAME
+
+TS_VERIFY_CTX_set_certs - set certificates for TS reponse verfication
```
Comment (FdaSilvaYY) [inline doc/man3/TS_VERIFY_CTX_set_certs.pod]: typo: verfication

Comment (mspncp) [reply doc/man3/TS_VERIFY_CTX_set_certs.pod]: two typos:  re**s**ponse ver**i**fication

File: doc/man3/TS_VERIFY_CTX_set_certs.pod [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+=head1 SYNOPSIS
+
+ #include <openssl/ts.h>
+
+ STACK_OF(X509) *TS_VERIFY_CTX_set_certs(TS_VERIFY_CTX *ctx,
+                                         STACK_OF(X509) *certs);
+
+=head1 DESCRIPTION
+
+The Time-Stamp protocol (TSP) is defined by RFC 3161. TSP is a protocol used to
```
Comment (mspncp) [inline doc/man3/TS_VERIFY_CTX_set_certs.pod]: In RFC 3161, three capital letters are used: "Time-Stamp **P**rotocol (TSP)"

File: doc/man3/TS_VERIFY_CTX_set_certs.pod [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+provide long term proof of the existence of a certain datum before a particular
+time. TSP defines a Time Stamping Authority (TSA) and an entity who shall make
+requests to the TSA. Usually the TSA is deemed as the server side and the
```
Comment (mspncp) [inline doc/man3/TS_VERIFY_CTX_set_certs.pod]: Maybe "deemed as" -> "denoted as"?

File: doc/man3/TS_VERIFY_CTX_set_certs.pod [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+requesting entity is deemed as the client.
+
+In TSP, when a server is sending a response to a client, the server normally
+needs to sign the response data - the TimeStampToken (TST) - with its private
+key. Then the client shall verify the received TST by server's certificate chain.
```
Comment (mspncp) [inline doc/man3/TS_VERIFY_CTX_set_certs.pod]: "by *the* server's"   (or "using the server's")

File: doc/man3/TS_VERIFY_CTX_set_certs.pod [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+
+TS_VERIFY_CTX_set_certs() is used to set the server's certificate chain when
+verifying a TST. B<ctx> is the verification context created in advance and
+B<certs> is a stack of B<X509> certificates in PEM format.
```
Comment (mspncp) [inline doc/man3/TS_VERIFY_CTX_set_certs.pod]: "~~in PEM format~~."  (The `X509` struct contains an internal representation of the certificate)

Comment (InfoHunter) [reply doc/man3/TS_VERIFY_CTX_set_certs.pod]: Hmmm, right. I was originally thinking the certificates in file should be in PEM format. But that's not this API's business to care about...

**Comment by InfoHunter** [comment]
All fixed in the latest commit

**Comment by InfoHunter** [comment]
Ping @openssl/committers

**Comment by InfoHunter** [comment]
Ahh, if this is back ported to 1.1.1, the function name should be changed back to the typo version...

**Comment by mspncp** [comment]
I didn’t follow all details, but could‘nt the corrected function be added in 1.1.1 without deprecation, and the pod file (with correct file name) would contain both functions in the NAME section?

**Comment by InfoHunter** [comment]
Depends on if the newly added function is considered as a new API or a bugfix

**Comment by InfoHunter** [comment]
Anyway, if we finally do this to 1.1.1, the man page would look like a little hilarious ;-)

**Comment by mspncp** [comment]
Ah, but an idea just occurred to me. I will post it on #9838 in a few minutes...

**Comment by mspncp** [comment]
>  I will post it on #9838 

I meant #9472 of course.

**Comment by InfoHunter** [comment]
> Please document also the compatibility macro Please `TS_VERIFY_CTS_set_certs()` and its deprecation, see #9472.

OK, will do that later

**Comment by InfoHunter** [comment]
`TS_VERIFY_CTS_set_certs()` has been documented to a `NOTES` sections, but I think it's also acceptable if documented in `HISTORY` section.

**Comment by InfoHunter** [comment]
Which one would you guys consider better?

**Comment by mspncp** [comment]
> This will definitely fail the test (doc-nits check) since #9472 has not been merged yet.

_Note: You could rebase your pull request on top of #9472 for the moment._

**Comment by mspncp** [comment]
> `TS_VERIFY_CTS_set_certs()` has been documented to a `NOTES` sections, but I think it's also acceptable if documented in `HISTORY` section.

I think a note will not be sufficient to satisfy find-doc-nits (after removing the entries from missingcrypo.txt and missingmacro.txt) you need a full blown documentation for the mispelled macro, with a `NAME`  and `SYNOPSIS` entry. And finally in the `HISTORY` comes the usual deprecation note. In order to try it out properly, you could rebase onto #9472, or you just wait until its merged. No hurry.

**Comment by mspncp** [comment]
> you need a full blown documentation for the mispelled macro,

In other words: the NAME section contains both the correct and the misspelled name, and the SYNOPSIS contains the prototype twice, misspelled and correct.

**Comment by mspncp** [comment]
#9472 has been merged now, you can rebase on master to pick it up.

Please be so kind to remove

- `TS_VERIFY_CTX_set_certs` from `util/missingcrypto.txt` and 
- `TS_VERIFY_CTS_set_certs` from `util/missingmacro.txt`

after rebasing.

**Comment by InfoHunter** [comment]
Sorry for the huge delay, too many meetings these days...

Rebased to #9472 . And the misspelled version `TS_VERIFY_CTS_set_certs` should also needed to be added into `util/other.syms` to satisfy `make doc-nits`

**Comment by InfoHunter** [comment]
Ping?

**Comment by InfoHunter** [comment]
It seems #9472 is not merged to 1.1.1, so this should not as well.

**Comment by mspncp** [comment]
> It seems #9472 is not merged to 1.1.1, so this should not as well.

Agreed.

**Comment by InfoHunter** [comment]
Any other suggestions on this?

**Comment by mspncp** [comment]
> ... and achieves exactly the same.

or simply:  ... and returns the same result.

**Comment by InfoHunter** [comment]
Haha, interesting

**Comment by InfoHunter** [comment]
Patched your patch in the latest commit

File: doc/man3/TS_VERIFY_CTX_set_certs.pod [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+B<certs> is a stack of B<X509> certificates.
+
+TS_VERIFY_CTS_set_certs() is a misspelled version of TS_VERIFY_CTX_set_certs()
+which takes the same parameters and returns the same result.
+
+=head1 NOTES
+
+The spelling of TS_VERIFY_CTX_set_certs() was corrected in OpenSSL 3.0.0.
+The misspelled version TS_VERIFY_CTS_set_certs() has been retained for
+compatibility reasons, but it is deprecated in OpenSSL 3.0.0.
```
Comment (mspncp) [inline doc/man3/TS_VERIFY_CTX_set_certs.pod]: Sorry for bothering you one more time: It occurred to me only now that you mention the deprecation in both the NOTES and HISTORY section. I think we normally put it into the HISTORY section, see `git grep`:


```make

~/openssl/openssl$ git --no-pager grep -n -B 3 'deprecated in' -- doc/man3 | grep head1
doc/man3/BIO_connect.pod-102-=head1 HISTORY
doc/man3/CONF_modules_free.pod-44-=head1 HISTORY
doc/man3/DH_generate_parameters.pod-145-=head1 HISTORY
doc/man3/DSA_generate_parameters.pod-112-=head1 HISTORY
doc/man3/ENGINE_add.pod-652-=head1 HISTORY
doc/man3/ERR_error_string.pod-68-=head1 HISTORY
doc/man3/ERR_load_crypto_strings.pod-41-=head1 HISTORY
doc/man3/ERR_remove_state.pod-36-=head1 HISTORY
doc/man3/OBJ_nid2obj.pod-177-=head1 HISTORY
doc/man3/OPENSSL_config.pod-73-=head1 HISTORY
doc/man3/RAND_add.pod-96-=head1 HISTORY
doc/man3/RAND_cleanup.pod-32-=head1 HISTORY
doc/man3/RSA_generate_key.pod-96-=head1 HISTORY
doc/man3/SSL_COMP_add_compression_method.pod-92-=head1 HISTORY
doc/man3/SSL_library_init.pod-43-=head1 HISTORY
doc/man3/TS_VERIFY_CTX_set_certs.pod-48-=head1 HISTORY
doc/man3/X509_VERIFY_PARAM_set_flags.pod-369-=head1 HISTORY

...

```


So my suggestion would be: move lines 39-42 down to the HISTORY section (replacing lines 50-51) and delete the NOTES section.

File: doc/man3/TS_VERIFY_CTX_set_certs.pod [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+
+=head1 RETURN VALUES
+
+TS_VERIFY_CTX_set_certs() returns the stack of B<X509> certificates the user
+passes in via parameter B<certs>.
+
+=head1 HISTORY
+
+TS_VERIFY_CTX_set_certs() was introduced in OpenSSL 3.0.0 as a typo-fix version
+of TS_VERIFY_CTS_set_certs(), which is also deprecated in 3.0.0.
```
Comment (mspncp) [inline doc/man3/TS_VERIFY_CTX_set_certs.pod]: See my previous comment.

**Comment by InfoHunter** [comment]
fixed

**Comment by InfoHunter** [comment]
Ping @openssl/omc

**Comment by InfoHunter** [comment]
This PR is stalled... Ping @openssl/omc again.

**Comment by FdaSilvaYY** [comment]
Ping @openssl/omc this one is very stall.
Should it be rebased to fix  util/missingcrypto.txt and  util/missingmacro.txt  conflict ?

**Comment by mspncp** [comment]
@InfoHunter can you rebase please? Then I'll reconfirm.

**Comment by mspncp** [comment]
> @InfoHunter can you rebase please? Then I'll reconfirm.

ping?

**Comment by mspncp** [comment]
Ups. I made a mistake when rebasing your pr. Hang on for a moment...

**Comment by mspncp** [comment]
> Ups. I made a mistake when rebasing your pr. Hang on for a moment...

I forgot to solve the conflicts in `util/missing*.txt' before pushing. Now it's done.

**Comment by InfoHunter** [comment]
> I forgot to solve the conflicts in `util/missing*.txt' before pushing. Now it's done.

Great, thanks. It seems that I missed the 'ping' from you ;-)

**Comment by InfoHunter** [comment]
But I am wondering how could you push commit to my branch?

**Comment by mspncp** [comment]
When you create a pull request, somewhere there is a check button

 * [x] Allow edits by maintainers

which is checked by default. If that is set, then commmitters can push to the pull request owner's branch in his own repository clone. (I guess that GitHub would refuse if I would  push to a different branch of the clone, but I never tried ;-) ).

All I need to do is to find out you pr branch name at the top of the page. In your case, I ran

```
git checkout pr-9838
git rebase -i
git push git@github.com:InfoHunter/openssl.git HEAD:doc-9472 [--force] 
```

**Comment by InfoHunter** [comment]
Good to know this feature, thanks

**Comment by bernd-edlinger** [comment]
I think you can also simply go to the files changed and edit in the web interface (I did it once and it
worked)

**Comment by richsalz** [comment]
> edit in the web interface

But those changes would be lost if the original author does a force-push (as I think most everyone does in this project).

**Comment by mspncp** [comment]
> > edit in the web interface
> 
> But those changes would be lost if the original author does a force-push 

The same holds for https://github.com/openssl/openssl/pull/9838#issuecomment-574514642.

> (as I think most everyone does in this project).

Hopefully you are wrong. At least I don't do that. IMHO there are two best practices to minimize the risk of such clashes:

* never do a force-push without need (i.e. unless you amended your last commit or rebased your branch).
* always run `git fetch --all` before force-pushing and watch git's output. Git will tell you exactly which remote branches were updated and whether they were force-pushed or added to. If your current branch is in the list, that should ring your alarm clock: you need to do a `git pull --rebase` first.

**Comment by richsalz** [comment]
> never do a force-push without need (i.e. unless you amended your last commit or rebased your branch).

And that is what I think is common practice.  WAY WAY WAY more than, say, someone else editing "your" commit  (Which seems to have happend, what, at most six times?)

**Comment by InfoHunter** [comment]
Pushed to master.

**Pull request closed without merging**